commons/zenoh-shm/tests/shm_buffer.rs

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright (c) 2025 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use zenoh_core::Wait;
+use zenoh_shm::api::{
+    buffer::{typed::Typed, zshmmut::ZShmMut},
+    provider::shm_provider::ShmProviderBuilder,
+};
+
+#[repr(C)]
+#[derive(zerocopy::KnownLayout, zerocopy::FromBytes)]
+struct SharedByteData {
+    data: [u8; 64],
+}
+
+fn make_shm_buffer() -> ZShmMut {
+    let provider = ShmProviderBuilder::default_backend(65536).wait().unwrap();
+    provider.alloc(64).wait().unwrap()
+}
+
+fn validate_raw_buffer_consistency(buffer: &mut ZShmMut) {
```

Comment:
Is this really necessary? Isn't memory on modern OS always set to 0 for security reasons?